CMakeLists.txt

```diff
@@ -34,6 +42,7 @@ if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24")
 endif()
 find_package(nlohmann_json 3.10.0 QUIET)
 if(NOT NLOHMANN_JSON_FOUND)
+  set(CabanaPD_INTERNAL_JSON ON)
```

Comment:
I think we also need to set the `JSON_Install` cmake variable to `ON` when `CabanaPD_INTERNAL_JSON` is set to ON in order to install the nlohmann_json target.